src/assets/token.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { getLisk32AddressFromAddress } from '@liskhq/lisk-cryptography';
+
+import {
+	MODULE_NAME_TOKEN,
+	MODULE_NAME_DPOS,
+	ADDRESS_LEGACY_RESERVE,
+	TOKEN_ID_LSK_MAINCHAIN,
+	LOCAL_ID_LSK,
+	RADIX_HEX,
+	MODULE_NAME_LEGACY,
+	LOCAL_ID_LENGTH,
+} from '../constants';
+
+import {
+	AccountEntry,
+	LegacyAccountEntry,
+	SupplySubstoreEntry,
+	TokenStoreEntry,
+	UserSubstoreEntry,
+	ModuleResponse,
+	LockedBalance,
+} from '../types';
+
+const nextLexicographicalOrder = (currentID: string) =>
+	(parseInt(currentID, RADIX_HEX) + 1).toString(RADIX_HEX).padStart(LOCAL_ID_LENGTH, '0');
+
+export const getLockedBalances = async (account: AccountEntry): Promise<LockedBalance[]> => {
+	let amount = 0;
+	for (const vote of account.dpos.sentVotes) {
+		amount += Number(vote.amount);
+	}
+
+	for (const unlockingObj of account.dpos.unlocking) {
+		amount += Number(unlockingObj.amount);
+	}
+
+	if (amount > 0) {
+		return [{ module: MODULE_NAME_DPOS, amount: String(amount) }];
+	}
+	return [];
+};
+
+export const createLegacyReserveAccount = async (
+	accounts: AccountEntry[],
+	legacyAccounts: LegacyAccountEntry[],
+): Promise<UserSubstoreEntry> => {
+	const legacyReserveAccount: any = accounts.find(
+		account => account.address.toString('hex') === ADDRESS_LEGACY_RESERVE.toString('hex'),
+	);
+	let legacyReserveAmount;
```

Comment:
`legacyReserveAmount` must be the balance for `legacyReserveAccount`. We can discuss the specifics offline.
```suggestion
	let legacyReserveAmount: bigint = legacyReserveAccount?.balance ?? BigInt('0');
```